HomeWork/src/main/java/streams/PetrolStation.java

```diff
@@ -0,0 +1,32 @@
+package streams;
+
+import java.util.concurrent.*;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class PetrolStation {
+    private double amount = 100;
+    private final Semaphore semaphore = new Semaphore(3);
+    private final Lock lock = new ReentrantLock();
+
+    public double getAmount() {
+        return amount;
+    }
+
+    public void setAmount(double amount) {
+        this.amount = amount;
+    }
+
+    public void doRefuel(double amountPetrol) throws InterruptedException {
+        semaphore.acquire();
+        lock.lock();
+        try {
+            amount -= amountPetrol;
```

Comment:
What if the initial `amount` of fuel becomes <= 0? How do you plan to handle this case?